### ISSUE TITLE
Acquiring next byte store may not change the state when shifting backwards

### ISSUE THREAD

**Issue by glukos** [OpenHFT/Chronicle-Bytes#452]
```
        if (adding > 0 && !bytesStore.inside(offset, checkSize0(adding - 1))) {
            acquireNextByteStore0(offset + adding - 1, false);
```
The logic above fails when `bytesStore.start()` is greater than `offset`, but the difference between start and `offset` is less than `adding`.
Real-life example is overwriting the header after we wrote a message.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.23.32](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.23.32), [BOM-2.23.127](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23.127)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.22.28](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.22.28), [BOM-2.22.162](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22.162)
